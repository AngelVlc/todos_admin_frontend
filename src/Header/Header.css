.Header {
    background-color: #282c34;
    display: flex;
    flex-direction: column;
    align-items: left;
    justify-content: center;
    font-size: calc(8px + 1vmin);
    color: white;
    padding-left: 50px;
}