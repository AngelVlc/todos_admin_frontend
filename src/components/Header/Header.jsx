import React, { useRef } from 'react';
import { useHistory } from 'react-router-dom';

export const Header = () => {
    const navBarMenuRef = useRef();
    const navBarBurguerRef = useRef();
    let history = useHistory();
    const userInfoStr = localStorage.getItem('userInfo');
    let userInfo = userInfoStr ? JSON.parse(userInfoStr) : null;

    const onLogoutClick = () => {
        localStorage.setItem('userInfo', null);
        history.push('/login');
    }

    const onToggleNavBar = (e) => {
        navBarMenuRef.current.classList.toggle('is-active');
        navBarBurguerRef.current.classList.toggle('is-active');
    }

    return (
        <nav className="navbar is-dark">
            <div className="navbar-brand">
                <span className="navbar-item is-size-3">To Dos</span>
                {userInfo &&
                    <div className="navbar-burger" ref={navBarBurguerRef} onClick={(e) => onToggleNavBar(e)}>
                        <span aria-hidden="true"></span>
                        <span aria-hidden="true"></span>
                        <span aria-hidden="true"></span>
                    </div>
                }
            </div>
            <div className="navbar-menu" ref={navBarMenuRef}>
                <div className="navbar-start">
                    {userInfo &&
                        <>
                            <span className="navbar-item" onClick={() => history.push('/')}>Home</span>
                            <span className="navbar-item" onClick={() => history.push('/lists')}>Lists</span>
                        </>
                    }
                    {userInfo?.isAdmin &&
                        <span className="navbar-item" onClick={() => history.push('/users')}>Users</span>
                    }
                </div>
            </div>
            {userInfo &&
                <div className="navbar-end">
                    <div className="navbar-item">
                        <div className="buttons">
                            <button className="button is-light is-small" data-testid="logOut" onClick={() => onLogoutClick()}>
                                <span>{userInfo.userName}</span>
                                <span className="icon is-small">
                                    <i className="fas fa-sign-out-alt"></i>
                                </span>
                            </button>
                        </div>
                    </div>
                </div>
            }
        </nav>
    )
}
