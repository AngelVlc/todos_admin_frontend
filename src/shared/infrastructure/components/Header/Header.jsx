import React, { useRef, useContext } from 'react';
import { AppContext } from '../../contexts';
import { useHistory } from 'react-router-dom';

export const Header = () => {
    const { auth } = useContext(AppContext)
    const navBarMenuRef = useRef();
    const navBarBurguerRef = useRef();
    let history = useHistory();

    const onLogoutClick = () => {
        history.push('/login');
    }

    const onToggleNavBar = (e) => {
        navBarMenuRef.current.classList.toggle('is-active');
        navBarBurguerRef.current.classList.toggle('is-active');
    }

    return (
        <nav className="navbar is-dark">
            <div className="navbar-brand">
                <span className="navbar-item is-size-3">To Dos</span>
                {auth.info &&
                    <div className="navbar-burger" ref={navBarBurguerRef} onClick={(e) => onToggleNavBar(e)}>
                        <span aria-hidden="true"></span>
                        <span aria-hidden="true"></span>
                        <span aria-hidden="true"></span>
                    </div>
                }
            </div>
            <div className="navbar-menu" ref={navBarMenuRef}>
                <div className="navbar-start">
                    {auth.info &&
                        <>
                            <span className="navbar-item" data-testid="goToRoot" onClick={() => history.push('/')}>Home</span>
                            <span className="navbar-item" data-testid="goToLists" onClick={() => history.push('/lists')}>Lists</span>
                        </>
                    }
                    {auth.info?.isAdmin &&
                        <span className="navbar-item" data-testid="goToUsers" onClick={() => history.push('/users')}>Users</span>
                    }
                </div>
            </div>
            {auth.info &&
                <div className="navbar-end">
                    <div className="navbar-item">
                        <div className="buttons">
                            <button className="button is-light is-small" data-testid="logOut" onClick={() => onLogoutClick()}>
                                <span>{auth.info.userName}</span>
                                <span className="icon is-small">
                                    <i className="fas fa-sign-out-alt"></i>
                                </span>
                            </button>
                        </div>
                    </div>
                </div>
            }
        </nav>
    )
}
