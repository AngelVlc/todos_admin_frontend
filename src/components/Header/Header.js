import React, { useContext } from 'react';
import { AppContext } from '../../contexts/AppContext';
import { userLoggedOut } from '../../actions/auth';
import './Header.css';

export const Header = () => {
    const { auth, authDispatch } = useContext(AppContext)

    const logoutClick = (authDispatch) => {
        authDispatch(userLoggedOut());
    }

    return (
        <header className="Header">
            <h1>To Dos Admin</h1>
            {auth.info &&
                <>
                    <h4>User: {auth.info.userName}</h4>
                    <button data-testid="logOut" onClick={() => logoutClick(authDispatch)}>Log out</button>
                </>
            }
        </header>
    )
}
