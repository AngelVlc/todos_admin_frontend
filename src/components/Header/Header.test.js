import React from 'react'
import { render, cleanup, fireEvent, wait } from '@testing-library/react'
import { Header } from './Header'
import { AppContext } from '../../contexts/AppContext'

const mockAuthDispatch = jest.fn()

const renderWithContext = (component, auth) => {
    const context = { authDispatch: mockAuthDispatch, auth }
    return {
        ...render(
            <AppContext.Provider value={context}>
                {component}
            </AppContext.Provider>)
    }
}

afterEach(cleanup)

it('should match the snapshot when the user is not logged in', () => {
    const { asFragment } = renderWithContext(<Header />, { info: null });

    expect(asFragment(<Header />)).toMatchSnapshot();
})

it('should match the snapshot when the user is logged in', () => {
    const { asFragment } = renderWithContext(<Header />, { info: { userName: 'user' } });

    expect(asFragment(<Header />)).toMatchSnapshot();
})

it('should do logout', async () => {
    const { getByTestId } = renderWithContext(<Header />, { info: { userName: 'user' } });

    await wait(() => {
        fireEvent.click(getByTestId('logOut'));
    })

    expect(mockAuthDispatch.mock.calls.length).toBe(1);
    const userLoggedIn = {
        type: 'USER_LOGGED_OUT'
    }
    expect(mockAuthDispatch.mock.calls[0][0]).toStrictEqual(userLoggedIn);
})