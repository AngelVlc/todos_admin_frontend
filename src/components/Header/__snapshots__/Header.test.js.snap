// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match the snapshot when the user is logged in 1`] = `
<DocumentFragment>
  <header
    class="Header"
  >
    <h1>
      To Dos Admin
    </h1>
    <h4>
      User: user
    </h4>
    <button
      data-testid="logOut"
    >
      Log out
    </button>
  </header>
</DocumentFragment>
`;

exports[`should match the snapshot when the user is not logged in 1`] = `
<DocumentFragment>
  <header
    class="Header"
  >
    <h1>
      To Dos Admin
    </h1>
  </header>
</DocumentFragment>
`;
